runtime/internal/services/autocli.go

```diff
@@ -0,0 +1,37 @@
+package services
+
+import (
+	"context"
+
+	autocliv1 "cosmossdk.io/api/cosmos/autocli/v1"
+	"cosmossdk.io/core/appmodule"
+)
+
+// AutoCLIQueryService implements the cosmos.autocli.v1.Query service.
+type AutoCLIQueryService struct {
+	autocliv1.UnimplementedQueryServer
+
+	moduleOptions map[string]*autocliv1.ModuleOptions
+}
+
+func NewAutoCLIQueryService(appModules map[string]appmodule.AppModule) *AutoCLIQueryService {
```

Comment:
how does this get handled for app_legacy? might be documented somewhere or is it automatic?

Reply:
As long as it's using runtime.App then it happens automatically.

Reply:
This didn't answer my question. How does app_legacy that doesn't use runtime enable this feature

Reply:
I thought the point of app legacy was to demonstrate manual wiring. App legacy can just use runtime but manually wired without depinject. I don't think we should be trying to maintain a totally runtime free path if we're actually building runtime.

Reply:
Would you like app legacy updated to use runtime like I'm proposing?

Another thing that could be done to integrate this better, although it is breaking, is to make types.AppModule an extension of the core AppModule interface and start making the module manager use core. That will make manual wiring simpler.

If we do any of these things though I'd prefer to tackle in separate PRs.

Reply:
> I don't think we should be trying to maintain a totally runtime free path if we're actually building runtime.

Agree with this sentiment, but also it would be nice if v0.47 had autocli w/o runtime. My preferred path would be:
- by v0.47: try to wire up autocli with current app_legacy, do not use/require runtime
- after v0.47 (release branch): update app_legacy to use runtime

Reply:
Runtime or no runtime, what will make it easier for app_legacy is a single base AppModule interface that this works off of. It's pretty easy to expose the here services as public but they depend on the app module set.

Reply:
its super unclear how runtime works without depinject. If you can demonstrate a demo we can migrate to that, but right now only having a depinject demo here to show users is unacceptable

Reply:
Would it be okay to address this in a follow up PR? All providers are public because of depinject changes so manually wiring runtime is 100% possible, but I'm not sure it's ergonomic and there are a few options of how to tweak things. I think that would be cleaner to explore in a separate PR if that's okay.

Reply:
works with me.